### PR TITLE
Upgrade Prometheus to v2.55.1

### PR DESCRIPTION
**Pull request by aknuds1**
#### What this PR does

Upgrade to mimir-prometheus synced with upstream v2.55.1 ([33bfe839e](https://github.com/grafana/mimir-prometheus/commit/33bfe839e)).

I propose this as an intermediary step before [upgrading to Prometheus v3](https://github.com/grafana/mimir/pull/9844), since that requires fixing a lot of currently failing tests.

Changes:

* [[FEATURE] PromQL: Add experimental `info` function](https://github.com/prometheus/prometheus/pull/14495)
* [PromQL: make sort_by_label stable](https://github.com/prometheus/prometheus/pull/14985)
* [Fix round function not honoring delayed name removal flag](https://github.com/prometheus/prometheus/pull/15250)

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +69,7 @@
 * [BUGFIX] Ingester: Fix race condition in per-tenant TSDB creation. #9708
 * [BUGFIX] Ingester: Fix race condition in exemplar adding. #9765
 * [BUGFIX] Ingester: Fix race condition in native histogram appending. #9765
+* [BUGFIX] PromQL: Fix round function not honoring delayed name removal flag. #9879
```
Comment (colega) [inline CHANGELOG.md:72]: There's no delayed name removal flag in Mimir, so this doesn't apply here.

Comment (colega) [reply CHANGELOG.md:72]: Although I see that this isn't accurate, actually the behaviour changes now, and `round()` now drops the metric name? This looks like a breaking change.

Comment (bboreham) [reply CHANGELOG.md:72]: `round()` always dropped the metric name; it was mistakenly changed to not do that in 2.55.0 and fixed in 2.55.1.

Comment (colega) [reply CHANGELOG.md:72]: Hum, what did we ship with 2.14?

Comment (aknuds1) [reply CHANGELOG.md:72]: Good point, let me compare to the previous release.

Comment (aknuds1) [reply CHANGELOG.md]: I fixed the changelog entry so it's accurate. I compared to v2.14.1 and can see that it's a fix versus that version.

Comment (aknuds1) [reply CHANGELOG.md]: @colega WDYT? Should it still be a BUGFIX, or a CHANGE instead?

File: CHANGELOG.md [hunk 2]
```diff
@@ -67,6 +69,7 @@
 * [BUGFIX] Ingester: Fix race condition in per-tenant TSDB creation. #9708
 * [BUGFIX] Ingester: Fix race condition in exemplar adding. #9765
 * [BUGFIX] Ingester: Fix race condition in native histogram appending. #9765
+* [BUGFIX] PromQL: `round` removes the metric name. #9879
```
Comment (colega) [inline CHANGELOG.md:72]:
```suggestion
* [BUGFIX] PromQL: `round` now removes the metric name again. #9879
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -56,6 +57,7 @@
 * [ENHANCEMENT] Ingester: improved lock contention affecting read and write latencies during TSDB head compaction. #9822
 * [ENHANCEMENT] Distributor: when a label value fails validation due to invalid UTF-8 characters, don't include the invalid characters in the returned error. #9828
 * [ENHANCEMENT] Ingester: when experimental ingest storage is enabled, do not buffer records in the Kafka client when fetch concurrency is in use. #9838 #9850
+* [ENHANCEMENT] PromQL: make sort_by_label stable. #9879
```
Comment (colega) [inline CHANGELOG.md:60]:
```suggestion
* [ENHANCEMENT] PromQL: make `sort_by_label` stable. #9879
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -31,6 +31,7 @@
 * [FEATURE] Ingester: Experimental support for ingesting out-of-order native histograms. This is disabled by default and can be enabled by setting `-ingester.ooo-native-histograms-ingestion-enabled` to `true`. #7175
 * [FEATURE] Distributor: Added `-api.skip-label-count-validation-header-enabled` option to allow skipping label count validation on the HTTP write path based on `X-Mimir-SkipLabelCountValidation` header being `true` or not. #9576
 * [FEATURE] Ruler: Add experimental support for caching the contents of rule groups. This is disabled by default and can be enabled by setting `-ruler-storage.cache.rule-group-enabled`. #9595
+* [FEATURE] PromQL: Add experimental `info` function. #9879
```
Comment (colega) [inline CHANGELOG.md:34]: Is it enabled by default? Should some flag be used for it?

File: pkg/frontend/querymiddleware/astmapper/parallel.go [hunk 5]
```diff
@@ -27,6 +27,8 @@ var NonParallelFuncs = []string{
 	// The following functions are not safe to parallelize.
 	"absent",
 	"absent_over_time",
+	// TODO: Find out whether should be parallelizable.
+	"info",
```
Comment (colega) [inline pkg/frontend/querymiddleware/astmapper/parallel.go:31]: Let's make clear that the comment applies only to this function, not the rest of the functions below.

```suggestion
	"info", // TODO: Find out whether can be parallelizable.
```

File: CHANGELOG.md (hunk 4 above)
Comment (aknuds1) [reply CHANGELOG.md]: Done, PTAL.

**Review by colega — APPROVED** [review]

**Pull request merged**